### PR TITLE
Changes FileSet Attachment Event to reflect initiating user (#1249).

### PR DESCRIPTION
**Pull request by bwatson78**
- .rubocop.yml: allows Hyrax-imported spec to pass rubocop linting.
- app/actors/hyrax/actors/create_with_files_actor.rb: passes initiating user to the AttachFilesToWorkJob.
- app/jobs/attach_files_to_work_job.rb: takes in and makes the initiating user the priority for the user associated for the File Attached change event.
- spec/*: changes expectations for arguments passed through as well as makes sure that the `FileSetAttachedEventJob` gets the right user.

### CODE REVIEW

**Comment by bwatson78** [comment]
No screenshots necessary.

File: app/actors/hyrax/actors/create_with_files_actor.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    # Creates a work and attaches files to the work
+    class CreateWithFilesActor < Hyrax::Actors::AbstractActor
+      # @param [Hyrax::Actors::Environment] env
+      # @return [Boolean] true if create was successful
+      def create(env)
+        uploaded_file_ids = filter_file_ids(env.attributes.delete(:uploaded_files))
+        files             = uploaded_files(uploaded_file_ids)
+        # get a current copy of attributes, to protect against future mutations
+        attributes        = env.attributes.clone
+        user = env&.user
```
Comment (devanshu-m) [inline app/actors/hyrax/actors/create_with_files_actor.rb:16]: Can you please add a comment here that this is new in this overwritten file

File: app/actors/hyrax/actors/create_with_files_actor.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      end
+
+      # @param [Hyrax::Actors::Environment] env
+      # @return [Boolean] true if update was successful
+      def update(env)
+        uploaded_file_ids = filter_file_ids(env.attributes.delete(:uploaded_files))
+        files             = uploaded_files(uploaded_file_ids)
+        # get a current copy of attributes, to protect against future mutations
+        attributes        = env.attributes.clone
+        user = env&.user
```
Comment (devanshu-m) [inline app/actors/hyrax/actors/create_with_files_actor.rb:30]: Same comment here please

File: app/actors/hyrax/actors/create_with_files_actor.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      def create(env)
+        uploaded_file_ids = filter_file_ids(env.attributes.delete(:uploaded_files))
+        files             = uploaded_files(uploaded_file_ids)
+        # get a current copy of attributes, to protect against future mutations
+        attributes        = env.attributes.clone
+        user = env&.user
+
+        validate_files(files, env) &&
+          next_actor.create(env) &&
+          attach_files(files, env.curation_concern, attributes, user)
```
Comment (devanshu-m) [inline app/actors/hyrax/actors/create_with_files_actor.rb:20]: Can you please add a comment here that user is the new arg

File: app/actors/hyrax/actors/create_with_files_actor.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      def update(env)
+        uploaded_file_ids = filter_file_ids(env.attributes.delete(:uploaded_files))
+        files             = uploaded_files(uploaded_file_ids)
+        # get a current copy of attributes, to protect against future mutations
+        attributes        = env.attributes.clone
+        user = env&.user
+
+        validate_files(files, env) &&
+          next_actor.update(env) &&
+          attach_files(files, env.curation_concern, attributes, user)
```
Comment (devanshu-m) [inline app/actors/hyrax/actors/create_with_files_actor.rb:34]: Same comment here please

**Comment by devanshu-m** [comment]
@bwatson78 any comment on the argument order being different in the two files?

**Comment by bwatson78** [comment]
@devanshu-m I didn't see that you requested that, but If you'd like me to add the comment, I will.

**Comment by devanshu-m** [comment]
@bwatson78 I meant wouldn't it cause issues if the order of args is different in one file as compared to the other? User is being passed as the fourth arg but when called/received, it is the third arg. Will this cause an issue? I understand optional args need to go towards the end, but I think this makes it necessary even when passing them, not just calling them.

**Review by devanshu-m — APPROVED** [review]

**Pull request merged**